include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,425 @@
+{
+	const struct stepper_api *api = (const struct stepper_api *)dev->api;
+
+	if (api->get_actual_position == NULL) {
+		return -ENOSYS;
+	}
+	return api->get_actual_position(dev, value);
+}
+
+/**
+ * @brief Set the absolute target position of the stepper
+ *
+ * @param dev pointer to the stepper motor controller instance
+ * @param value target position to set in micro_steps
+ * @param async Pointer to a valid and ready to be signaled struct
+ *              k_poll_signal. If changing the target position
+ *              triggers stepper movement, this can be used to await
+ *              the end of the transaction. (Note: can be left NULL)
+ *
+ * @retval -ENOSYS If not implemented by device driver
+ * @retval       0 Success
+ */
+__syscall int32_t stepper_set_target_position(const struct device *dev, const int32_t value,
+					      struct k_poll_signal *async);
+
+static inline int32_t z_impl_stepper_set_target_position(const struct device *dev,
+							 const int32_t value,
+							 struct k_poll_signal *async)
+{
+	const struct stepper_api *api = (const struct stepper_api *)dev->api;
+
+	if (api->set_target_position == NULL) {
+		return -ENOSYS;
+	}
+	return api->set_target_position(dev, value, async);
+}
+
+/**
+ * @brief Check if the stepper motor is currently moving
+ *
+ * @param dev pointer to the stepper motor controller instance
+ * @param is_moving Pointer to a boolean to store the moving status of the stepper motor
+ *
+ * @retval -ENOSYS If not implemented by device driver
+ * @retval       0 Success
+ */
+__syscall int32_t stepper_is_moving(const struct device *dev, bool *is_moving);
+
+static inline int32_t z_impl_stepper_is_moving(const struct device *dev, bool *is_moving)
+{
+	const struct stepper_api *api = (const struct stepper_api *)dev->api;
+
+	if (api->is_moving == NULL) {
+		return -ENOSYS;
+	}
+	return api->is_moving(dev, is_moving);
+}
+
+/**
+ * @brief Enable constant velocity mode for the stepper with a given velocity
+ *
+ * @details activate constant velocity mode with the given velocity in micro_steps_per_second.
+ * If velocity > 0, motor shall be set into motion and run incessantly until and unless stalled or
+ * stopped using some other command, for instance, motor_enable(false).
+ *
+ * @param dev pointer to the stepper motor controller instance
+ * @param direction The direction to set
+ * @param value The velocity to set in steps per second where one step is dependent on the current
+ * microstepping resolution:
+ * > 0: Enable constant velocity mode with the given velocity in a given direction
+ * 0: Disable constant velocity mode
+ *
+ * @retval -ENOSYS If not implemented by device driver
+ * @retval       0 Success
+ */
+__syscall int32_t stepper_enable_constant_velocity_mode(const struct device *dev,
+							const enum stepper_direction direction,
+							const uint32_t value);
+
+static inline int32_t z_impl_stepper_enable_constant_velocity_mode(
```

Comment:
return values are normally `int`, not `int32_t`, I think these should all be changed